streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -477,7 +484,7 @@ private boolean checkMetadataVersions(final int minReceivedMetadataVersion,
      * @return map from repartition topic to its partition info
      */
     private Map<TopicPartition, PartitionInfo> prepareRepartitionTopics(final Map<Integer, TopicsInfo> topicGroups,
-                                                                           final Cluster metadata) {
+                                                                           final Cluster metadata) throws TaskAssignmentException {
```

Comment:
Could we update the meta comment to explain when a `TaskAssignmentException` is thrown?

Reply:
`TaskAssignmentException` is not a checked exception and it's just a curtesy declaration... (I can also remove `throws TaskAssignmentException` if you prefer).

Don't see any need to document anything further -- the code makes it clear (in fact, this method does not even throw the exception itself, but it just bubbles up from `internalTopicManager.makeReady`.